lighthouse-core/computed/page-dependency-graph.js

```diff
@@ -394,6 +394,8 @@ class PageDependencyGraph {
     const networkNodeOutput = PageDependencyGraph.getNetworkNodeOutput(networkRecords);
     const cpuNodes = PageDependencyGraph.getCPUNodes(processedTrace);
 
+    // TODO: Remove this usage of `findMainDocument`. This assumption isn't true if there was a JS redirect.
+    // ...
```

Comment:
This file seems to depend `mainDocumentRequest` actually being the "first document request" in the case of a JS redirect. It might require a larger refactor in a different PR to eliminate the usage of `NetworkAnalyzer.findMainDocument` here.